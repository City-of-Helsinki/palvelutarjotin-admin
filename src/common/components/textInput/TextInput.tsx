import React from 'react';

import InputWrapper, { InputWrapperProps } from './InputWrapper';
import styles from './inputWrapper.module.scss';

export type TextInputProps = {
  type?: string;
} & InputWrapperProps;

const TextInput: React.FC<TextInputProps> = ({ type = 'text', ...props }) => {
  const {
    defaultValue,
    labelledBy,
    disabled,
    id,
    readOnly,
    onBlur,
    onChange,
    placeholder,
    value,
  } = props;
  return (
    <InputWrapper {...props}>
      <input
        type={type}
        className={styles.input}
        defaultValue={defaultValue}
        aria-labelledby={labelledBy}
        disabled={disabled}
        id={id}
        readOnly={readOnly}
        onBlur={onBlur}
        onChange={onChange}
        placeholder={placeholder}
        value={value}
      />
    </InputWrapper>
  );
};

export default TextInput;
