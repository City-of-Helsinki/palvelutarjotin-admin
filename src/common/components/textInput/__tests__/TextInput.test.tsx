import { render } from '@testing-library/react';
import React from 'react';

import TextInput from '../TextInput';

const textInputProps = {
  id: 'hdsInput',
  labelText: 'HDS input field',
  placeholder: 'A placeholder text',
};

describe('<TextInput /> spec', () => {
  it('renders the component', () => {
    const { asFragment } = render(<TextInput {...textInputProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders the component with tooltip', () => {
    const { asFragment } = render(
      <TextInput
        tooltipText="Tooltip text"
        tooltipLabel="Tooltip label"
        {...textInputProps}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
