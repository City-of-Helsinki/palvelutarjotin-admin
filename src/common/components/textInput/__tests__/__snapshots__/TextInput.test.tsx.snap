// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> spec renders the component 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <label
      class="label "
      for="hdsInput"
    >
      HDS input field
    </label>
    <div
      class="inputWrapper"
    >
      <input
        class="input"
        id="hdsInput"
        placeholder="A placeholder text"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`<TextInput /> spec renders the component with tooltip 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <label
      class="label "
      for="hdsInput"
    >
      HDS input field
    </label>
    <button
      aria-label=""
      class="Tooltip-module_buttonTooltip__1u5aY"
      title=""
      type="button"
    >
      <span
        aria-hidden="true"
      >
        <svg
          class="Tooltip-module_iconTooltip__2BS7K"
          height="100%"
          viewBox="0 0 23 23"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.363 9.421a.898.898 0 1 1-1.794 0 3.91 3.91 0 0 1 1.148-2.774 3.91 3.91 0 0 1 2.774-1.148 3.91 3.91 0 0 1 2.774 1.148 3.91 3.91 0 0 1 1.148 2.774 3.91 3.91 0 0 1-1.148 2.775 3.898 3.898 0 0 1-1.875 1.044v.388a.899.899 0 1 1-1.798 0v-1.184c0-.492.402-.894.899-.894.587 0 1.12-.24 1.505-.624a2.124 2.124 0 0 0 0-3.01 2.124 2.124 0 0 0-3.01 0 2.124 2.124 0 0 0-.623 1.505zm2.038 8.075a.895.895 0 0 1-.9-.894c0-.497.403-.9.9-.9h.198c.497 0 .9.403.9.9a.895.895 0 0 1-.9.894h-.198zm.099 3.706a9.656 9.656 0 0 0 6.86-2.843 9.656 9.656 0 0 0 2.842-6.859 9.675 9.675 0 0 0-2.843-6.864A9.675 9.675 0 0 0 11.5 1.794a9.694 9.694 0 0 0-6.864 2.842A9.694 9.694 0 0 0 1.794 11.5a9.675 9.675 0 0 0 2.842 6.86 9.675 9.675 0 0 0 6.864 2.842zm8.13-1.573A11.454 11.454 0 0 1 11.5 23c-3.177 0-6.05-1.288-8.134-3.37A11.468 11.468 0 0 1 0 11.5a11.48 11.48 0 0 1 3.366-8.134A11.48 11.48 0 0 1 11.5 0c3.172 0 6.05 1.288 8.13 3.366A11.464 11.464 0 0 1 23 11.5c0 3.172-1.288 6.05-3.37 8.13z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </button>
    <div
      class="inputWrapper"
    >
      <input
        class="input"
        id="hdsInput"
        placeholder="A placeholder text"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
